### PR TITLE
Show test outputs and duration on the test analysis page

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-651](https://issues.redhat.com//browse/TRT-651) 

- Refactors the test analysis page to show data as we have it, instead of staying stuck at "Loading..." because the slowest parts haven't finished yet. This should be a pretty big improvement to that page's experience.
- Show at most the last 10 failure outputs from a test on the test analysis page, making it easier to troubleshoot simple failures without having to click through multiple times to Prow.
- Plot the average test duration by day

Screenshot 2022-11-01 at 10-08-32 Sippy 4 12 Tests sig-arch events should not repeat pathologically

### CODE REVIEW

**Comment by stbenjam** [comment]
/hold

It's a little slow, need to figure out how to improve it

File: pkg/api/tests.go [hunk 1]
```diff
@@ -34,6 +34,40 @@ func PrintTestsDetailsJSONFromDB(w http.ResponseWriter, release string, testSubs
 	RespondWithJSON(http.StatusOK, w, responseStr)
 }
 
+func PrintTestOutputsFromDB(dbc *db.DB, release, test string, filters *filter.Filter, quantity int) ([]apitype.TestOutput, error) {
```
Comment (dgoodwin) [inline pkg/api/tests.go:37]: Suggest GetTestOutputsFromDB or Load or something like that, we don't actually write to the http response here.

**Review by dgoodwin — COMMENTED** [review]
This is a beauty. Just a couple nits.

File: pkg/sippyserver/server.go [hunk 2]
```diff
@@ -463,6 +463,76 @@ func (s *Server) jsonTestBugsFromDB(w http.ResponseWriter, req *http.Request) {
...
+		})
+		return
+	}
+
+	filters, err := filter.ExtractFilters(req)
+	if err != nil {
+		api.RespondWithJSON(http.StatusInternalServerError, w, map[string]interface{}{
+			"code":    http.StatusInternalServerError,
+			"message": "error processing filter options",
+		})
```
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/sippyserver/server.go:486]: I always forget these, thanks

**Comment by stbenjam** [comment]
Should be good to go

**Comment by dgoodwin** [comment]
/lgtm
/override ci/prow/e2e

Which is busted.

**Pull request merged**